src/app/page.tsx

```diff
@@ -1,10 +1,12 @@
-import { Timer } from "@/app/ui/timer/timer";
-import { getHoffmanBetterOneCupMethodSteps } from "@/app/lib/recipies/drip/hoffman";
+import {useState} from "react";
+import {ClientUi} from "@/app/ui/clientUi";
+import {getRawRecipes} from "@/app/lib/recipes/utils";
 
 export default function Home() {
+  let rawRecipesMap = getRawRecipes()
```

Comment:
ok,  still the same issue:
<img width="1659" alt="image" src="https://github.com/jakub-filistynski/coffee-brew-timer/assets/42801480/9d776a0d-6b2f-4887-ba37-8a1c368f4858">

Reply:
I am not super sure but: https://github.com/reactjs/rfcs/blob/main/text/0188-server-components.md#capabilities--constraints-of-server-and-client-components

<img width="1166" alt="image" src="https://github.com/jakub-filistynski/coffee-brew-timer/assets/42801480/3852f9c3-6a33-4685-80b9-df4fa33cc85c">

Reply:
And solution would be to fetch data in server component which calls client component render. If we try to make root component "use client" it is calling server function on initial render, as described in error.

Reply:
Also I am no sure if I get "server-only data sources" right, but solution seems to be making root component "use server", call all server functions there, and (as we cannot use state in it) wrap all client components into ClientUI and call it from root component, which kinda resolves what we already have

Reply:
aaand this error is called when I "use server" on top of whole file, and call it anywhere in client component directly

Reply:
Ok, a solution would be to fetch data in useEffect

Reply:
But it still shows same error:
 `Server Functions cannot be called during initial render. This would create a fetch waterfall. Try to use a Server Component to pass data to Client Components instead.` So IMO to avoid this waterfall client components need to be wrapped (?)

Reply:
although calling it from useEffect like that 
<img width="743" alt="image" src="https://github.com/jakub-filistynski/coffee-brew-timer/assets/42801480/16176105-a343-49ba-9c6a-635139b78687">
causes warnings Assignments to the 'rawRecipesMap' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.

Reply:
Ok so proper way seems to look like that:
Root component (use server) => Root client component with state (use client) ->
```export function ClientUi() {
  const [rawRecipesMap, setRawRecipesMap] = useState(new Map())

  useEffect(() => {
    const fetchRecipes = async (): Promise<Map<string, RecipeSchema>> => {
      const rawRecipes = await getRawRecipes()
      setRawRecipesMap(rawRecipes)
    }
    fetchRecipes()
  },[])
```

Reply:
Ok I dont get what and when happened, but now root with "use client" works without errors 🥲

Reply:
I am so confused